pkg/runtime/cache/namespaces.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	informersv1 "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	k8scache "k8s.io/client-go/tools/cache"
+
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+)
+
+// namespaceInfo contains annotations ACK controllers care about
+type namespaceInfo struct {
+	// services.k8s.aws/default-region Annotation
+	defaultRegion string
+	// services.k8s.aws/owner-account-id Annotation
+	ownerAccountID string
+}
+
+// GetDefaultRegion returns the default region value
+func (n *namespaceInfo) GetDefaultRegion() string {
+	if n == nil {
+		return ""
+	}
+	return n.defaultRegion
+}
+
+// GetOwnerAccountID returns the namespace owner Account ID
+func (n *namespaceInfo) GetOwnerAccountID() string {
+	if n == nil {
+		return ""
+	}
+	return n.ownerAccountID
+}
+
+// NamespaceCache is reponsible of keeping track of namespaces
+// annotations, and caching those related to the ACK controller.
+type NamespaceCache struct {
+	sync.RWMutex
+
+	log logr.Logger
+	// Namespace informer
+	informer k8scache.SharedInformer
+	// namespaceInfos maps namespaces names to their known namespaceInfo
+	namespaceInfos map[string]*namespaceInfo
+}
+
+// NewNamespaceCache makes a new NamespaceCache from a
+// clientset.Interface and a logr.Logger
+func NewNamespaceCache(kc clientset.Interface, log logr.Logger) *NamespaceCache {
+	sharedInformer := informersv1.NewNamespaceInformer(
+		kc,
+		informerResyncPeriod,
+		k8scache.Indexers{},
+	)
+	return &NamespaceCache{
+		informer:       sharedInformer,
+		log:            log.WithName("NamespaceCache"),
+		namespaceInfos: make(map[string]*namespaceInfo),
```

Comment:
how is the `namespaceInfos` map going to be populated on startup of the service controller?